### DISCUSSION TITLE
Getting index.html for slug in nextjs 14  in production

### DISCUSSION THREAD

**Discussion by pforprince** [vercel/next.js#62730] [App Router]
Hi, im using nextjs for my projects, i want to read the value of [username] and use it, it is working fine in local but when im working on production, the dynamic value is always index.html

Below is the code 

![code](https://github.com/vercel/next.js/assets/64328117/c1161c26-9817-49f3-8ff1-e8d42e3ee215)

Below is screenshot from production 

![WhatsApp Image 2024-03-01 at 20 11 42](https://github.com/vercel/next.js/assets/64328117/f2c3c047-4a26-4e40-9e07-07003434e1ae)

**Comment by icyJoseph** [comment]
Woa, that's weird, are you using export mode? or perhaps rewrites? do you observe the same if you do `next build && next start` on your machine? (a so called local production build)?

**Comment by pforprince** [comment]
@icyJoseph same is there even in build running on local. I'm not using rewrites

**Reply by icyJoseph** [reply]
what Nextjs version do you have? and is it export mode?

**Reply by pforprince** [reply]
next version - "14.0.3". What is export mode?

**Reply by icyJoseph** [reply]
That's when the app is built as a Static HTML site, but you'd know that because you have to explicitly say that in your next.config file.


I'm not sure then, I'd need to know more about the app, or if you could set a stackblitz reproduction example (just type next.new into your browser URL bar, and enter) save the demo and share the link back here

**Reply by pforprince** [reply]
Would it be nearly possible to connect online for this one? I'm not sure if the provided snippet gonna give the error. Im sure it'll work since im can't copy-paste the whole code.

**Reply by pforprince** [reply]
local build is working fine now, but on prod im still facing the issue

**Reply by pforprince** [reply]
Production bug 

![WhatsApp Image 2024-03-02 at 14 22 59](https://github.com/vercel/next.js/assets/64328117/1d006d8c-a84d-4f22-b7b8-776b68f36412)

**Reply by pforprince** [reply]
![WhatsApp Image 2024-03-02 at 14 34 41](https://github.com/vercel/next.js/assets/64328117/8a8dbfc2-c5f3-4343-b4dc-2b733911b344)

**Reply by pforprince** [reply]
Could anyone help?

**Reply by icyJoseph** [reply]
How does the link that takes you there look like?

**Reply by pforprince** [reply]
![image](https://github.com/vercel/next.js/assets/64328117/a6706a77-7178-4ef1-8719-e0bdbfa2dca3)

**Reply by icyJoseph** [reply]
Right, I think the issue must be that the username route is intercepting the request for the root page. 

Do you have an `app/page.tsx` file? If not, make one, temporarily, with some dummy content. Does that fix the issue?

**Reply by pforprince** [reply]
i've a page.tsx inside app directory. It's still not working

**Reply by icyJoseph** [reply]
This is going nowhere without code, I'm sorry. Are you sure you can't make a Stackblitz reproduction? 

Are you perhaps using trailing slashes?

**Reply by pforprince** [reply]
Thanks a lot man, it worked. I don't know but there was a rewrite inside vercel.json that was causing the issue.

**Reply by icyJoseph** [reply]
Nice! Yeah it had to be something like that.

By the way, imperative navigation, such as using router push, rather than an anchor tag or, next/link, can cause poor user experience, and negatively affect SEO.
